packages/remix-react/components.tsx

```diff
@@ -1298,6 +1293,182 @@ export function useSubmitImpl(key?: string): SubmitFunction {
+    //     5. Create an entry with namex and x, and append it to entry list.
+    //     6. Create an entry with namey and y, and append it to entry list.
+    //     7. Continue.
+    if (
+      isImageButtonElement(field) &&
+      field === submitter &&
+      field[SELECTED_COORDINATE]
+    ) {
+      let prefix = field.name ? `${field.name}.` : "";
+      let { x, y } = field[SELECTED_COORDINATE];
+      entryList.push([`${prefix}x`, String(x)]);
+      entryList.push([`${prefix}y`, String(y)]);
+      continue;
+    }
+
+    //  3. If the field is a form-associated custom element, then perform the entry construction algorithm given field and entry list, then continue.
+    //  4. If either the field element does not have a name attribute specified, or its name attribute's value is the empty string, then continue.
+    //  5. Let name be the value of the field element's name attribute.
+    let { name } = field;
+    if (!name) continue;
+
+    //  6. If the field element is a select element, then for each option element in the select element's list of options whose selectedness is true and that is not disabled, create an entry with name and the value of the option element, and append it to entry list.
+    if (isSelectElement(field)) {
+      for (let option of field.selectedOptions) {
+        if (!option.disabled) entryList.push([name, option.value]);
+      }
+    }
+    //  7. Otherwise, if the field element is an input element whose type attribute is in the Checkbox state or the Radio Button state, then:
+    //     1. If the field element has a value attribute specified, then let value be the value of that attribute; otherwise, let value be the string "on".
+    //     2. Create an entry with name and value, and append it to entry list.
+    else if (
+      isInputElement(field) &&
+      ["radio", "checkbox"].includes(field.type)
+    ) {
+      if (field.checked) entryList.push([name, field.value]);
+    }
+    //  8. Otherwise, if the field element is an input element whose type attribute is in the File Upload state, then:
+    //     1. If there are no selected files, then create an entry with name and a new File object with an empty name, application/octet-stream as type, and an empty body, and append it to entry list.
+    //     2. Otherwise, for each file in selected files, create an entry with name and a File object representing the file, and append it to entry list.
+    else if (isInputElement(field) && field.type === "file") {
+      if (field.files?.length) {
+        for (let file of field.files) {
+          entryList.push([name, file]);
+        }
+      } else {
+        entryList.push([name, ""]);
+      }
+    }
+    //  9. Otherwise, if the field element is an input element whose type attribute is in the Hidden state and name is an ASCII case-insensitive match for "_charset_":
```

Comment:
missed this too 😅  ... i'm not actually sure if browser implement this though, should verify

Reply:
implemented